scripts/validate_unwanted_patterns.py

```diff
@@ -58,6 +58,7 @@
     "_iLocIndexer",
     # TODO(3.0): GH#55043 - remove upon removal of ArrayManager
     "_get_option",
+    "_to_numpy_dtype_inference",
```

Comment:
Since the original method is in core, probably can remove the leading `_`

Reply:
done